File: ruby/simulate_payment.rb
require_relative 'lightning_network_rpc_client'

class SimulatePayment
  def run
    invoice = alice.add_invoice(value: 1000000000000, memo: 'Macdonald')
    puts "Invoice from alice: #{invoice.payment_request}"
    bob.pay_invoice(payment_request: invoice.payment_request, timeout_seconds: 120).each do |state|
      puts state.status
    end
  end

  def bob
    @bob ||= LightningNetworkRpcClient.new("127.0.0.1:10002", node_name = "bob")
  end

  def alice
    @alice ||= LightningNetworkRpcClient.new("127.0.0.1:10001", node_name = "alice")
  end
end


if $PROGRAM_NAME == __FILE__
  s = SimulatePayment.new
  s.run
end

File: python/simulate_payment.py
from logging import debug
from config import LNDConfiguration
import router_pb2 as routerrpc, lightning_pb2 as lnrpc

def main() -> None:
    alice = LNDConfiguration(node_name="alice", GRPC_host="127.0.0.1:10001")
    bob = LNDConfiguration(node_name="bob", GRPC_host="127.0.0.1:10002")

    lightning_stub = alice.lightning_stub()
    request_invoice = lnrpc.Invoice(memo="Mac donlad", value=1000)

    response = lightning_stub.AddInvoice(request_invoice, metadata=[('macaroon', alice.macaroon)])

    routing_stub = bob.router_stub()
    request = routerrpc.SendPaymentRequest(
        payment_request=response.payment_request,
        fee_limit_sat=10000,
        allow_self_payment = True,
        timeout_seconds=120
    )

    for response in routing_stub.SendPaymentV2(request, metadata=[('macaroon', bob.macaroon)]):
        print(response)

if __name__ == '__main__':
    main()